app/src/main/java/de/simon/dankelmann/bluetoothlespam/Adapters/AdvertisementSetCollectionExpandableListViewAdapter.kt

```diff
@@ -71,6 +72,18 @@ class AdvertisementSetCollectionExpandableListViewAdapter internal constructor(
 
         holder.label?.text = advertisementSet.title
         holder.label?.setTextColor(textColor)
+        
+        // Set checkbox state from the model
+        holder.checkbox?.isChecked = advertisementSet.isChecked
+        
+        // Add listener to update the model when checkbox state changes
+        holder.checkbox?.setOnClickListener { view ->
```

Comment:
You can use `Checkbox.setOnCheckedChangeListener` to have simpler code.

Reply:
Nevermind. I gave it a try in #366, and when using onCheckedChangeListener the UI jumps around a lot. On the surface, it seems as if things were being reordered, or like views are recycled and not properly initialised. Not sure why.

So let’s keep onClickListener for now.